src/main/java/com/zendesk/maxwell/producer/MaxwellRedisProducer.java

```diff
@@ -129,9 +143,7 @@ public void push(RowMap r) throws Exception {
 				break;
 			} catch (Exception e) {
 				if (e instanceof JedisConnectionException) {
-					logger.warn("lost connection to server, recreating redis connection...", e);
-					jedis.disconnect();
-					initRedisConnection();
+					logger.warn("lost connection to server, will try again with another connection from pool", e);
```

Comment:
does the pool throw away the connection and re-establish?  or is there a possibility of running out of pools?  

It'd be wonderful to establish an integration test for these behaviors, but that would blow out the scope of the PR greatly.  In lieu of that, is there an easy way to trigger `JedisConnectionException` manually and verify that the connection pool does the right thing and throws the connection away?

Reply:
Yes it does. Since Jedis is a Closeable object, when the try block finishes, the connection is returned to the pool. As the pool is configured to `setTestOnBorrow(true)` the connection is recreated if is no good anymore.

I've tested here throwing the JedisConnectionException occasionally (time % 10), I could see a lot of errors (expected) but no downtime, all messages succeeded in the next atempt.

And I've measured the pool health with active connections and idle connections. The pool starts with 1/0 with a regular usage (binlog streaming). After starting a new bootstrap task, the pool stays with 2/0, when the bootstrap task finishes it stays with 1/1 till the idle connection drops.

IMHO, everything working great here.

Reply:
great, thanks for the dilligence